lib/src/command/token_add.dart

```diff
@@ -40,17 +48,11 @@ class TokenAddCommand extends PubCommand {
         throw DataException('Insecure package repository could not be added.');
       }
 
-      final token = await stdinPrompt('Enter secret token:')
-          .timeout(const Duration(minutes: 15));
-      if (token.isEmpty) {
-        usageException('Token is not provided.');
+      if (envVar?.isNotEmpty == true) {
+        await _addTokenFromEnv(hostedUrl);
+      } else {
+        await _addTokenFromStdin(hostedUrl);
```

Comment:
I think it should be an error to have an empty envVar, but probably better to check it inside `_addTokenFromEnvVar`.
```suggestion
      if (envVar == null) {
        await _addTokenFromStdin(hostedUrl);
      } else {
        await _addTokenFromEnvVar(hostedUrl);
      }
```